src/sqlfluff/cli/formatters.py

```diff
@@ -22,10 +22,25 @@
 
 
 def split_string_on_spaces(s: str, line_length: int = 100) -> List[str]:
-    """Split a string into lines based on whitespace."""
+    """Split a string into lines based on whitespace.
+
+    For short strings the functionality is trivial.
+    >>> split_string_on_spaces("abc")
+    ['abc']
+
+    For longer sections it will split at an appropriate point.
+    >>> split_string_on_spaces("abc def ghi", line_length=7)
+    ['abc def', 'ghi']
+
+    After splitting, multi-space sections should be intact.
+    >>> split_string_on_spaces("a '   ' b c d e f", line_length=11)
+    ["a '   ' b c", 'd e f']
+    """
     line_buff = []
     str_buff = ""
```

Comment:
Not part of this PR, but I notice that this function builds a string through repeated concatenation, which can be slow (n^2) behavior. Should we consider using `io.StringIO` to build the string instead? AFAIK, that's the canonical Python way to build a string incrementally.

Reply:
Good point - I've just done some experiments, and it's a little tricky because we also need to call `len()` on it to do the stacking and that doesn't work easily on `StringIO`. Good idea though.